Unity/opentok/holo/holo/Capturer.mm

```diff
@@ -0,0 +1,373 @@
+//        NSLog(@"[holo]: Capturer %p capturer New frame captured (width %d and height %d)", self, videoFrame.width, videoFrame.height);
+
+    if ([videoFrame.buffer isKindOfClass:[RTC_OBJC_TYPE(RTCCVPixelBuffer) class]]) {
+        CVPixelBufferRef buffer = ((RTC_OBJC_TYPE(RTCCVPixelBuffer) *)videoFrame.buffer).pixelBuffer;
+        OTVideoOrientation orientation = OTVideoOrientationUp;
+        switch ([videoFrame rotation]) {
+            case RTCVideoRotation_0:
+                orientation = OTVideoOrientationUp;
+                break;
+            case RTCVideoRotation_90:
+                orientation = OTVideoOrientationLeft;
+                break;
+            case RTCVideoRotation_180:
+                orientation = OTVideoOrientationDown;
+                break;
+            case RTCVideoRotation_270:
+                orientation = OTVideoOrientationRight;
+                break;
+
+            default:
+                orientation = OTVideoOrientationUp;
+                break;
+        }
+        NSData *data = nil;
+        if ([videoFrame isAugmented]) {
+            data = [NSData dataWithBytes:[videoFrame augmentingData] length:[videoFrame augmentingDataSize]];
+//            NSLog(@"[holo]: Capturer %p capturer augmented data size is %lu", self, static_cast<size_t>(data.length));
+        }
+        // TODO: Review the way we build the timestamp we provide through CMTimeMake (not sure is correct).
+        [_videoCaptureConsumer consumeImageBuffer:buffer
+                                      orientation:orientation
+                                        timestamp: CMTimeMake([videoFrame timeStampNs], CMTimeScale(NSEC_PER_SEC))
+                                         metadata:data];
+
+    }
+}
+
+- (void)setupListenerBlocks {
+}
+
+- (void)showCapturerError:(OTError*)error {
+    dispatch_async(dispatch_get_main_queue(), ^{
+        UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"[holo] Capturer"
+                                                                                 message:[NSString stringWithFormat:
+                                                                                          @"Capturer failed with error : %@", error.description]
+                                                                          preferredStyle:UIAlertControllerStyleAlert];
+        UIAlertAction *actionOk = [UIAlertAction actionWithTitle:@"Ok"
+                                                           style:UIAlertActionStyleDefault
+                                                         handler:nil];
+        [alertController addAction:actionOk];
+        [[[UIApplication sharedApplication] delegate].window.rootViewController
+         presentViewController:alertController
+         animated:YES completion:nil];
+    });
+}
+
+- (NSString*)captureSessionPreset {
+    return _capturePreset;
+}
+
+- (void)setCaptureSessionPreset:(NSString*)preset {
```

Comment:
i dont think this function is used.

Reply:
It might be called by the OpenTok video client SDK.

Reply:
Removed it.